src/foundry/client-esm/applications/api/dialog.d.mts

```diff
@@ -307,4 +298,16 @@ type InferButtonReturnTypes<Options extends { buttons?: DialogV2.Button<any>[] }
       ? Callback | string
       : never;
 
+type InferConfirmReturnTypes<
```

Comment:
Not a huge fan of the `Infer` prefix. Yes that's what it's doing but I'd prefer if it was just something like `ConfirmReturnType`. Not `Types` because even though it's a union it's all considered one type, singular.